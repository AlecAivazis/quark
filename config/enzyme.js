// external imports
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import browserEnv from 'browser-env'

// add a request animation frame polyfill
import raf from 'raf'
raf.polyfill()

// enable the react 16 adapter for enzyme
Enzyme.configure({ adapter: new Adapter() })

// add various browser mocks
browserEnv()
