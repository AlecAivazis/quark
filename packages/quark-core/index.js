export * from './components';
import * as _styles from './styles';
export { _styles as styles };