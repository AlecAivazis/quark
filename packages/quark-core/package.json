{
    "name": "quark-core",
    "version": "1.0.6",
    "module": "index.js",
    "main": "index.js",
    "dependencies": {
        "react-native-web": "0.3.4",
        "svgs": "^3.1.1"
    },
    "peerDependencies": {
        "react": "^16.3.0"
    }
}
