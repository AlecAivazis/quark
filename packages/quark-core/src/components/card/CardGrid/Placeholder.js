// @flow
// external imports
import React from 'react'
import { View, Text, StyleSheet } from 'react-native'

type Props = {
    style: number,
    unused?: {}
}

const CardPlaceholder = ({ style, ...unused }: Props) => (
    <View style={[styles.container, style]} />
)

const styles = StyleSheet.create({
    container: {
        flex: 1
    }
})

export default CardPlaceholder
