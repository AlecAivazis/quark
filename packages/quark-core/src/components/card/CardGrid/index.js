// @flow
// external imports
import React from 'react'
import { ScrollView, View, StyleSheet } from 'react-native'
import _ from 'lodash'
// local imports
import CardPlaceholder from './Placeholder'

type Props = {
    style: number,
    children: React.Element<*>,
    nCols?: number,
    padding?: number,
    unused: {}
}

const CardGrid = ({
    style,
    children,
    nCols = 3,
    padding = 40,
    ...unused
}: Props) => {
    // the style for cards
    const card = {
        marginBottom: padding,
        marginRight: padding
    }
    // the style of the last card
    const lastCard = {
        marginBottom: padding
    }

    // get the list of children
    const cards = React.Children.toArray(children)

    return (
        <ScrollView {...unused} style={[styles.container, style]}>
            {_.chunk(cards, nCols).map((row, i) => (
                <View style={styles.row} key={`row-${i}`}>
                    {row.map((child, i) => {
                        // the style to apply to the card
                        const cardStyle = i === row.length - 1 ? lastCard : card
                        return (
                            <View
                                style={[styles.card, cardStyle]}
                                key={`card-${i}`}
                            >
                                {child}
                            </View>
                        )
                    })}
                    {[...new Array(nCols - row.length)].map((_, i) => (
                        <CardPlaceholder
                            style={styles.cardWrapper}
                            key={`placeholder-${i}`}
                        />
                    ))}
                </View>
            ))}
        </ScrollView>
    )
}

const styles = StyleSheet.create({
    container: {
        backgroundColor: '#F9F9F9',
        padding: 40
    },
    row: {
        flex: 1,
        flexDirection: 'row',
        justifyContent: 'space-around'
    },
    cardWrapper: {
        flex: 1,
        marginRight: 40,
        marginBottom: 40
    },
    card: {
        flex: 1
    }
})

export default CardGrid
