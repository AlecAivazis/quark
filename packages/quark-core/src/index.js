// @flow
export * from './components'
export * as styles from './styles'
