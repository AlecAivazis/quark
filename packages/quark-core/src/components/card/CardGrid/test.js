// external imports
import React from 'react'
import { Text } from 'react-native'
// local imports
import CardGrid from './'
import { Card } from 'quark-core/components/card'

import renderer from 'react-test-renderer'

// the subject of our test
const Subject = ({ ...unused }) => (
    <CardGrid {...unused}>
        <Card>
            <Text>hello</Text>
        </Card>
        <Card>
            <Text>hello</Text>
        </Card>
        <Card>
            <Text>hello</Text>
        </Card>
        <Card>
            <Text>hello</Text>
        </Card>
        <Card>
            <Text>hello</Text>
        </Card>
    </CardGrid>
)

it('defaults to 3 columns', () => {
    const rendered = renderer.create(<Subject />).toJSON()
    expect(rendered).toMatchSnapshot()
})

it('supports specifying number of columns', () => {
    const rendered = renderer.create(<Subject nCols={2} />).toJson()
    expect(rendered).toMatchSnapshot()
})
