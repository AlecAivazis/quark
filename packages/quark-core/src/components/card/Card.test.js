// external imports
import React from 'react'
import { Text } from 'react-native'
import { mount } from 'enzyme'
// local imports
import CardGrid from './'
import { Card } from 'quark-core'

import renderer from 'react-test-renderer'

it('renders', () => {
    const rendered = mount(
        <Card>
            <Text>hello</Text>
        </Card>
    )

    expect(rendered).toMatchSnapshot()
})
