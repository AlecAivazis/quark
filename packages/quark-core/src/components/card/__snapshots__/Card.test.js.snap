// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<Card>
  <GetTheme>
    <View
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#EAEAEE",
          "borderRadius": 2,
          "borderStyle": "solid",
          "borderWidth": 1,
          "flex": 1,
          "shadowColor": "#EAEAEE",
          "shadowOffset": Object {
            "height": 2,
            "width": 2,
          },
          "shadowOpacity": 0.54,
          "shadowRadius": 4,
        }
      }
    >
      <div
        className="rn-alignItems-1oszu61 rn-borderTopLeftRadius-1iymjk7 rn-borderTopRightRadius-s2skl2 rn-borderBottomRightRadius-l5bh9y rn-borderBottomLeftRadius-101sy47 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-5kkj8d rn-borderRightWidth-13l2t4g rn-borderBottomWidth-qklmqi rn-borderLeftWidth-1ljd8xs rn-boxSizing-deolkf rn-display-6koalj rn-flex-13awgt0 rn-flexGrow-1m1wadx rn-flexShrink-1awmn5t rn-flexDirection-eqz5dr rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-zIndex-1lgpqti"
        style={
          Object {
            "backgroundColor": "rgba(255,255,255,1)",
            "borderBottomColor": "rgba(234,234,238,1)",
            "borderLeftColor": "rgba(234,234,238,1)",
            "borderRightColor": "rgba(234,234,238,1)",
            "borderTopColor": "rgba(234,234,238,1)",
            "boxShadow": "2px 2px 4px rgba(234,234,238,0.54)",
          }
        }
      >
        <Text>
          <div
            className="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-homxoj rn-display-1471scf rn-font-1lw9tu2 rn-fontFamily-10u92zi rn-fontSize-1b43r93 rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
            dir="auto"
          >
            hello
          </div>
        </Text>
      </div>
    </View>
  </GetTheme>
</Card>
`;
